pkg/rpc/auth.go

```diff
@@ -143,8 +143,20 @@ func (a kvAuth) authenticate(ctx context.Context) (roachpb.TenantID, error) {
 		//
 		// Is this a connection from another SQL tenant server?
 		if security.IsTenantCertificate(clientCert) {
-			// Incoming connection originating from a tenant SQL server. Let through.
-			// The other server is able to use any of this server's RPCs.
+			// Incoming connection originating from a tenant SQL server.
+			tid, err := tenantFromCommonName(clientCert.Subject.CommonName)
+			if err != nil {
+				return roachpb.TenantID{}, err
+			}
+			// Verify that our peer is a service for the same tenant
+			// as ourselves (we don't want to allow tenant 123 to
+			// serve requests for a client coming from tenant 456).
+			if tid != a.tenant.tenantID {
+				return roachpb.TenantID{}, authErrorf("this tenant (%v) cannot serve requests from a server for tenant %v", a.tenant.tenantID, tid)
+			}
+
+			// We return an unset tenant ID, to bypass authorization checks:
```

Comment:
what advantage does bypassing authorization checks here give us?

Reply:
Successfully authenticated node-node RPCs and root/node client principals have access to all the RPCs.  This has been true through the history of CockroachDB.  

We could change this by introducing a more granular authorization model for RPCs, but that sounds like a new feature/project which would deserve some thinking.